File: 프로그래머스/unrated/176963. 추억 점수/추억 점수.swift
import Foundation

func solution(_ name:[String], _ yearning:[Int], _ photo:[[String]]) -> [Int] {
    var result : [Int] = []
    for i in photo{
        var sum = 0
        for j in i{
            if let index = name.firstIndex(of: j) {
                sum += yearning[index]
            }
        }
        result.append(sum)
    }
    
    return result
}

File: 프로그래머스/unrated/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    dic = dict(zip(name,yearning))
    result = []
    for i in photo:
        sum = 0        
        for j in i:
            if dic.get(j) == None:
                print("없는값")
            else:
                sum += dic[j]
        result.append(sum)

    return result